mne/realtime/epochs.py

```diff
@@ -314,6 +314,7 @@ def _process_raw_buffer(self, raw_buffer):
         verbose = 'ERROR'
         sfreq = self.info['sfreq']
         n_samp = len(self._raw_times)
+        prev_samp = int(self._find_events_kwargs['min_samples'] - 1)
```

Comment:
Sorry for a bit of a delay. If the buffers are as follow [... 0 0 0][ 5 5 5 5 5 0 0 ... ] the code misses the onset of the event but reports the offset and stays stuck waiting for a pair for the offset. This does indeed solve the problem of two events but introduces a new bug where it's possible to miss some epochs. I added a test to test this and you can check that it gets stuck on 'Waiting for epoch 1'.